File: Practica1Ruby/pkg/Practica1Ruby-0.0.1/lib/FactoriaCarretera.rb
require('./CarreraCarretera.rb');
require('./BicicletaCarretera.rb');
require('./FactoriaCarreraYBicicleta.rb');

public
class FactoriaCarretera < FactoriaCarreraYBicicleta

	public
	def crearCarrera(numBicicletas)
		participantes = Array.new
    
    bicicleta = BicicletaCarretera.new(0,false)
    
        
    for i in (0..numBicicletas-1)
      bicicleta_aniadir = crearBicicleta(bicicleta) 
      if (i%11==0)
        bicicleta_aniadir.retirarse = true
      else
        bicicleta_aniadir.retirarse = false
      end
      participantes << bicicleta_aniadir
    end
    carreraCreada = CarreraCarretera.new(participantes)
    return carreraCreada
	end

	def crearBicicleta(bicicleta)
		return bicicleta.clone
	end

  public
	def initialize()
    
	end
  
  public_class_method :new
  
end

File: Practica1/src/practica1/FactoriaCarretera.java
package practica1;

import java.util.ArrayList;

public class FactoriaCarretera implements FactoriaCarreraYBicicleta {


    public FactoriaCarretera() {
    }
        
        
    
    @Override
    public CarreraCarretera crearCarrera( int numBicicletas) {
        ArrayList<BicicletaCarretera> participantes = new ArrayList<>();
        boolean retirarse = false;
        
        for (int i=0; i<numBicicletas; i++){
            
            if (i%11==0)
                retirarse = true;
            else
                retirarse = false;
            
            participantes.add(crearBicicleta(i,retirarse));
        }
            CarreraCarretera carreraCreada = new CarreraCarretera(participantes);
            
            return carreraCreada;
    }

    @Override
    public BicicletaCarretera crearBicicleta(int numBicicleta, boolean retirarse) {
            BicicletaCarretera bicicletaCreada = new BicicletaCarretera(numBicicleta, retirarse);
            return bicicletaCreada;
    }

}